File: JavaScript/8js 1/cookie.js
/*
|--------------------------------------------------------------------------
| Who ate the cookie?
|--------------------------------------------------------------------------
|
*/

function cookie(x) {
   let name = typeof x == "string" ? "Zach" : typeof x == "number" ? "Monica" : "the dog";
   return `Who ate the last cookie? It was ${name}!`;
}


function cookie(x) {
   return `Who ate the last cookie? It was ${
      { string: "Zach", number: "Monica" }[typeof x] || "the dog"
   }!`;
}


function cookie(x) {
   let resultString = "Who ate the last cookie? It was ";
   switch (typeof x) {
      case "string":
         return (resultString += "Zach!");
      case "number":
         return (resultString += "Monica!");
      default:
         return (resultString += "the dog!");
   }
}


console.log(cookie("Ryan"), "Who ate the last cookie? It was Zach!");
console.log(cookie(26), "Who ate the last cookie? It was Monica!");
console.log(cookie(2.3), "Who ate the last cookie? It was Monica!");
console.log(cookie(true), "Who ate the last cookie? It was the dog!");

File: Python/8py 1/cookie.py
"""
|--------------------------------------------------------------------------
| Who ate the cookie?
|--------------------------------------------------------------------------
|
"""

def cookie(x):
   name = "Zach" if type(x) == str else "Monica" if type(x) == int or type(x) == float else "the dog"
   return f"Who ate the last cookie? It was {name}!"


def cookie(x):
   return "Who ate the last cookie? It was %s!" % {str:"Zach", float:"Monica", int:"Monica"}.get(type(x), "the dog")


def cookie(x):
   return 'Who ate the last cookie? It was %s!' % ('the dog', 'Monica', 'Zach')[
      (type(x) in (int, float)) + 2 * (type(x) == str)]


print(cookie("Ryan"), "Who ate the last cookie? It was Zach!")
print(cookie(2.3), "Who ate the last cookie? It was Monica!")
print(cookie(26), "Who ate the last cookie? It was Monica!")
print(cookie(True), "Who ate the last cookie? It was the dog!")